### DISCUSSION TITLE
alternative to jsesc usage in the server runtime

### DISCUSSION THREAD

**Discussion by redabacha** [remix-run/remix#3854] [Proposals]
~~currently, the server runtime uses [`jsesc`](https://github.com/mathiasbynens/jsesc) to serialize the server handoff [here](https://github.com/remix-run/remix/blob/97999d02493e8114c39d48b76944069d58526e8d/packages/remix-server-runtime/serverHandoff.ts#L6) which is extremely slow compared to `JSON.stringify` (almost 10x slower with larger sets of data). as far as i understand this is primarily done to prevent xss attacks but it's very hard sometimes to cut down on loader data size such as when trying to render components based on data received from a headless cms.~~

~~hence, i'm wondering if it's possible to use an alternative approach to [`jsesc`](https://github.com/mathiasbynens/jsesc)?~~

~~or if that's not possible, can we add an escape hatch / option for a remix application to use `JSON.stringify` anyway when serializing the server handoff (with very clear and bold warning that it's subject to xss vulnerabilities, kinda like how react does it with `dangerouslySetInnerHTML`)? it would save a significant amount of cpu time which is generally quite precious when running code on the edge.~~

**this has been resolved by #3889 available in remix 1.7.0+**

**Comment by brophdawg11** [comment]
This discussion is being closed as part of the Remix v3 repository cleanup.

Remix v1/v2 was [upstreamed into React Router v7](https://remix.run/blog/wake-up-remix), so if this proposal is still relevant, please reopen or continue the discussion in the [React Router](https://github.com/remix-run/react-router) repository.
